haystack/components/rankers/sentence_transformers_diversity.py

```diff
@@ -218,17 +273,85 @@ def _greedy_diversity_order(self, query: str, documents: List[Document]) -> List
 
         return ranked_docs
 
+    def _embed_and_normalize(self, query, texts_to_embed):
+        # Calculate embeddings
+        doc_embeddings = self.model.encode(texts_to_embed, convert_to_tensor=True)  # type: ignore[attr-defined]
+        query_embedding = self.model.encode([self.query_prefix + query + self.query_suffix], convert_to_tensor=True)  # type: ignore[attr-defined]
+
+        # Normalize embeddings to unit length for computing cosine similarity
+        if self.similarity == "cosine":
+            doc_embeddings /= torch.norm(doc_embeddings, p=2, dim=-1).unsqueeze(-1)
+            query_embedding /= torch.norm(query_embedding, p=2, dim=-1).unsqueeze(-1)
+        return doc_embeddings, query_embedding
+
+    def _maximum_margin_relevance(
+        self, query: str, documents: List[Document], lambda_threshold: float = 0.5, top_k: Optional[int] = None
+    ) -> List[Document]:
```

Comment:
```suggestion
        self, query: str, documents: List[Document], lambda_threshold: float, top_k: int
    ) -> List[Document]:
```

My impression is that this method will never receive `None` values for `lambda_threshold` and `top_k`. Right?